state-sync/diem-data-client/src/diemnet/mod.rs

```diff
@@ -226,22 +222,46 @@ impl DiemNetDataClient {
                 Ok(Response::new(context, response))
             }
             Err(err) => {
-                // TODO(philiphayes): convert `err` into `ResponseError`
-                let error_type = ResponseError::InvalidData;
-                self.notify_bad_response(id, peer, &request, error_type);
-                Err(err)
+                // convert network error and storage service error types into
+                // data client errors. also categorize the error type for scoring
+                // purposes.
+                let (client_err, err_type) = match err {
+                    storage_service_client::Error::RpcError(err) => match err {
```

Comment:
nit: these all return the same `ErrorType` so could you refactor that out to reduce the duplication.